master/internal/task/allocation.go

```diff
@@ -509,8 +525,12 @@ func (a *Allocation) SetResourcesAsDaemon(
 }
 
 // HandleSignal handles an external signal to kill or terminate the allocation.
-func (a *Allocation) HandleSignal(ctx *actor.Context, msg actor.Message) {
-	switch msg {
+func (a *Allocation) HandleSignal(ctx *actor.Context, msg AllocationSignalWithReason) {
```

Comment:
eventually in allocation.Cleanup when this circles around and we send the exit event and it is logged, it prints a message like "Container for ... has exited". can you change all the 'container' messages to 'allocation' so this all makes more sense when it is printed. (they are already sort of wrong)